src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_SwitchExpression.cs

```diff
@@ -153,47 +172,106 @@ bool implicitConversionExists(BoundExpression expression, TypeSymbol type)
                 }
             }
 
-            private static BoundStatement ConstructThrowSwitchExpressionExceptionHelperCall(SyntheticBoundNodeFactory factory, BoundExpression unmatchedValue)
+            private BoundExpression LowerSwitchExpressionForRelease(BoundConvertedSwitchExpression node)
+            {
+                var outerVariables = ArrayBuilder<LocalSymbol>.GetInstance();
+                var loweredSwitchGoverningExpression = _localRewriter.VisitExpression(node.Expression);
+                var sideEffectsBuilder = ArrayBuilder<BoundExpression>.GetInstance();
+
+                BoundDecisionDag decisionDag = ShareTempsIfPossibleAndEvaluateInput(
+                    node.GetDecisionDagForLowering(_factory.Compilation, out LabelSymbol? defaultLabel),
+                    loweredSwitchGoverningExpression, sideEffectsBuilder.Add, out BoundExpression savedInputExpression);
+
+                Debug.Assert(savedInputExpression != null);
+
+                // lower the decision dag.
+                (ImmutableArray<BoundStatement> loweredDag, ImmutableDictionary<SyntaxNode, ImmutableArray<BoundStatement>> switchSections) =
+                    LowerDecisionDag(decisionDag);
+
+                if (_whenNodeIdentifierLocal is not null)
+                {
+                    outerVariables.Add(_whenNodeIdentifierLocal);
+                }
+
+                // Lower each switch expression arm
+                var switchArmsBuilder = ArrayBuilder<BoundLoweredSwitchExpressionArm>.GetInstance(node.SwitchArms.Length + (defaultLabel is null ? 0 : 1));
+                foreach (BoundSwitchExpressionArm arm in node.SwitchArms)
+                {
+                    _factory.Syntax = arm.Syntax;
+                    ImmutableArray<BoundStatement> statements = switchSections[arm.Syntax];
+                    BoundExpression loweredValue = _localRewriter.VisitExpression(arm.Value);
+
+                    // Lifetime of these locals is expanded to the entire switch body, as it is possible to
+                    // share them as temps in the decision dag.
+                    outerVariables.AddRange(arm.Locals);
+                    switchArmsBuilder.Add(new BoundLoweredSwitchExpressionArm(
+                        arm.Syntax, arm.Locals, statements.Add(_factory.Label(arm.Label)), loweredValue));
```

Comment:
Doing this instead of a separate label field because we want "statements" emitted before the label while "value prologue" produced in the spiller should be added afterwards. So this saves two fields and simplify lots of places especially spiller. However, now we call RecordLabel for this instead of RecordBranch which is done for is-expressions that I'm not quite sure about.